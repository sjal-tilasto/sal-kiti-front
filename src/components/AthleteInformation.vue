<template>
  <div>
    <b-row>
      <b-col>
        <h1>{{ athlete.first_name }} {{ athlete.last_name }}</h1>
      </b-col>
    </b-row>
    <b-row>
      <b-col>
        <h2 class="bg-sal-orange">{{ $t("athlete.info") }}</h2>
      </b-col>
    </b-row>
    <b-row>
      <b-col>
        <b-alert v-if="errors.main" variant="danger" show>
          <ul>
            <li v-for="e in errors.main" v-bind:key="e">{{ e }}</li>
          </ul>
        </b-alert>
      </b-col>
    </b-row>
    <b-row v-if="loadingAthlete">
      <b-col>
        <b-spinner label="Loading..."></b-spinner>
      </b-col>
    </b-row>
    <b-row v-else>
      <b-col cols="6" md="3">
        <dl>
          <dt>{{ $t("first_name") }}</dt>
          <dd v-if="athlete.first_name">{{ athlete.first_name }}</dd>
        </dl>
      </b-col>
      <b-col cols="6" md="3">
        <dl>
          <dt>{{ $t("last_name") }}</dt>
          <dd v-if="athlete.last_name">{{ athlete.last_name }}</dd>
        </dl>
      </b-col>
      <b-col cols="6" md="3">
        <dl>
          <dt>{{ $t("athlete.sport_id") }}</dt>
          <dd v-if="athlete.sport_id">{{ athlete.sport_id }}</dd>
        </dl>
      </b-col>
      <b-col cols="6" md="3">
        <dl>
          <dt>{{ $t("athlete.club") }}</dt>
          <dd v-if="athlete.organization_info">
            {{ athlete.organization_info.name }}
          </dd>
        </dl>
      </b-col>
    </b-row>
  </div>
</template>

<script>
/**
 * Displays basic information for a single athlete
 */
import apiGet from "../mixins/ApiGet";

export default {
  name: "AthleteInformation",
  mixins: [apiGet],
  data() {
    return {
      athlete: {},
      errors: {},
      loadingAthlete: true
    };
  },
  mounted() {
    this.getAthlete(this.$route.params.athlete_id);
  }
};
</script>

<style scoped></style>
