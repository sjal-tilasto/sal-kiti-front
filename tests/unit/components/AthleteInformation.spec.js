import { createLocalVue, shallowMount } from "@vue/test-utils";
import flushPromises from "flush-promises";
import AthleteInformation from "@/components/AthleteInformation.vue";
import BootstrapVue from "bootstrap-vue";
import Vuex from "vuex";

const localVue = createLocalVue();
localVue.use(BootstrapVue);
const state = { editMode: false, user: { is_staff: false } };
const store = new Vuex.Store({ state });

describe("AthleteInformation.vue", () => {
  it("renders athlete information", async () => {
    const $route = {
      params: { athlete_id: 1 }
    };
    const wrapper = shallowMount(AthleteInformation, {
      localVue,
      store,
      mocks: {
        $route
      }
    });
    await flushPromises();
    expect(wrapper.element).toMatchSnapshot();
  });

  it("render error message on error", async () => {
    const $route = {
      params: { athlete_id: 99 }
    };
    const wrapper = shallowMount(AthleteInformation, {
      localVue,
      store,
      mocks: {
        $route
      }
    });
    await flushPromises();
    expect(wrapper.text()).toContain("import.error.unknown");
  });
});
