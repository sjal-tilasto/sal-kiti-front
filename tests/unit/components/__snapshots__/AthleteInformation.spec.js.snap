// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AthleteInformation.vue renders athlete information 1`] = `
<div>
  <b-row-stub
    tag="div"
  >
    <b-col-stub
      tag="div"
    >
      <h1>
        Master Archer
      </h1>
    </b-col-stub>
  </b-row-stub>
   
  <b-row-stub
    tag="div"
  >
    <b-col-stub
      tag="div"
    >
      <h2
        class="bg-sal-orange"
      >
        athlete.info
      </h2>
    </b-col-stub>
  </b-row-stub>
   
  <b-row-stub
    tag="div"
  >
    <b-col-stub
      tag="div"
    >
      <!---->
    </b-col-stub>
  </b-row-stub>
   
  <b-row-stub
    tag="div"
  >
    <b-col-stub
      tag="div"
    >
      <b-spinner-stub
        label="Loading..."
        role="status"
        tag="span"
        type="border"
      />
    </b-col-stub>
  </b-row-stub>
</div>
`;
